otcextensions/sdk/dms/v1/message.py

```diff
@@ -9,23 +9,112 @@
 # WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 # License for the specific language governing permissions and limitations
 # under the License.
+from openstack import exceptions
 from openstack import resource
-from openstack import _log
 
-from otcextensions.sdk.dms.v1 import _base
 
-_logger = _log.setup_logging('openstack')
+class Message(resource.Resource):
 
+    resource_key = 'message'
+    base_path = '/queues/%(queue_id)s/groups/%(group_id)s/messages'
 
-class Message(_base.Resource):
+    allow_list = True
+
+    _query_mapping = resource.QueryParameters(
+        'max_msgs', 'time_wait', 'ack_wait'
+    )
+
+    #: Queue id
+    queue_id = resource.URI('queue_id')
+    #: Group ID (part of URL for listing)
+    group_id = resource.URI('group_id')
+
+    # Properties
+    #: Attributes
+    attributes = resource.Body('attributes', type=dict)
+    #: Message body
+    body = resource.Body('body')
+    #: Error info
+    error = resource.Body('error')
+    #: Error code
+    error_code = resource.Body('error_code')
+    #: Message handler (ID)
+    handler = resource.Body('handler', alternate_id=True)
+    #: State (0 - success)
+    state = resource.Body('state', type=int)
+
+    def _collect_attrs(self, attrs):
+        """ Save remaining attributes under "attributes" attribute
```

Comment:
can you please adjust a bit description? It sounds like which witch watch which watches.
What are these remaining attributes? Sub attributes of parameter attribute?

Reply:
yes, it is. It's a more or less copy from upstream, where it is like that and I prefer to keep as much from original as possible. Anyway, it is a purely internal function